applications/nrf_desktop/src/modules/dfu.c

```diff
@@ -356,10 +358,35 @@ static void store_dfu_data_chunk(void)
 	if (sync_offset - store_offset < store_size) {
 		store_size = sync_offset - store_offset;
 	}
-	if ((store_size > sizeof(uint32_t)) &&
-	    ((store_size % sizeof(uint32_t)) != 0)) {
+	if ((store_size > (4 * sizeof(uint32_t))) &&
+	    ((store_size % (4 * sizeof(uint32_t))) != 0)) {
 		/* Required by some flash drivers. */
-		store_size = (store_size / sizeof(uint32_t)) * sizeof(uint32_t);
+		store_size = (store_size / (4 * sizeof(uint32_t))) * 4 * sizeof(uint32_t);
+	}
+	if (store_size < (4 * sizeof(uint32_t))) {
```

Comment:
Can we rely on: `flash_get_write_block_size`? That would make implementation more generic

Reply:
Also I would make flash aligning as a separate commit.